node/lib/util/open.js

```diff
@@ -216,18 +229,30 @@ Opener.prototype.isOpen = co.wrap(function *(subName) {
  * is *unset*; since this operation is expensive, we cannot do it automatically
  * each time a submodule is opened.
  *
- * @param {String} subName
+ * @param {String}  subName
+ * @param {boolean} bare
  * @return {NodeGit.Repository}
  */
-Opener.prototype.getSubrepo = co.wrap(function *(subName) {
+Opener.prototype.getSubrepo = co.wrap(function *(subName, bare=false) {
     if (!this.d_initialized) {
         yield this._initialize();
     }
     let subRepo = this.d_subRepos[subName];
     if (undefined !== subRepo) {
         return subRepo;  // it was found
     }
-    if (this.d_openSubs.has(subName)) {
+    if (bare) {
```

Comment:
We only need to fully open the submodule if (a) there is a conflict and (b) the meta repo isn't bare (or whatever it is that you're using to decide whether to allow the user to resolve conflicts vs just instantly failing).  The rest of the time, half-opening it is fine.

Reply:
we can support interactive merge with half open later